<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Xilong Zhao, Junwei Han" />

<meta name="date" content="2024-09-16" />

<title>CITMIC User Guide</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">CITMIC User Guide</h1>
<h4 class="author">Xilong Zhao, Junwei Han</h4>
<h4 class="date">2024-09-16</h4>


<div id="TOC">
<ul>
<li><a href="#introduce" id="toc-introduce">Introduce</a></li>
<li><a href="#refenrence" id="toc-refenrence">Refenrence</a></li>
</ul>
</div>

<div id="introduce" class="section level2">
<h2>Introduce</h2>
<p>This vignette illustrates how to easily use the <strong>CITMIC
package</strong>. Here, we present a network-based computational
approach, <strong>CTMIC</strong>, estimate cell infiltration by
integrating gene transcriptomic data and individualized cell crosstalk
network.</p>
<p>This network-based method major parts:</p>
<ul>
<li><ul>
<li><p>Constructing network and performing randomization. In our method,
Here we collected a total of 86 cell types and 4171 GO-BPs. Among them,
the gene characterization sets for various cell types came from 12
sources, including published cell signature sets (Bindea et al. [1],
Charoentong et al. [2], Danaher et al. [3], Davoli et al. [4], He et
al. [5], Rooney et al. [6], Tirosh et al. [7]) and the TME cell
estimation methods (MCPcounter [8], EPIC [9], ImmuCellAI [10], TIDE [11]
and xCell [12]). The molecular function gene sets were downloaded from
C5 Gene sets derived from the GO biological process ontology in the
Molecular Signatures Database (MSigDB). We constructed a Cell-GO
bipartite network for each sample. We defined an edge between a Cell and
a GO term if they have at least a common gene, and give the weight of
this edge that is calculated by the Jaccard index and Gene Expression
Profile(GEP). Next, we proposed to construct a Cell crosstalk network
for each sample. The information from the Cell-GO bipartite network was
used to evaluate the extent of interaction between two Cells as the
edge’s weight. We consider that the extent of interaction between two
Cells will be stronger when there are more GO terms they target in
common and the more related they are to these GO terms. Then, we used
the eigenvector centrality measure to calculate how central each Cell is
in this individualized Cell crosstalk network. Finally, cell
infiltration in the patient dataset was obtained by normalizing the
centrality scores of the cells.</p>
<p>This package provides the <code>GetData_CITMIC</code> function to
return example data set and environment variables, such as the gene
expression matrix and so on.</p></li>
</ul></li>
</ul>
<hr />
<p>##Example calculate cell infiltration.</p>
<p>The function <code>CITMIC</code> is used to calculate the Cell
infiltration (lnScore). This function requires users to input the gene
expression matrix, the <em>weighted</em> specifies whether to create a
weighted graph for the cell crosstalk network, the <em>base</em> for the
standardized log base of data, the <em>damping</em> is the restart
probability of random walk and the <em>cl.cores</em> is a number of CPU
cores applied to this task. <em>cell.type</em> preset the relevant cell
type (e.g. if the solid tumor tissue does not contain ‘HSC’, it is
better to remove it when we preset it). We recommend using as input the
gene expression matrix normalized by log2(fpkm+1). An example data is
stored in the package.</p>
<p>The commands are as follows:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co">#Detect the Number of CPU Cores (Select the number of cl.cores to apply to CITMIC based on detectCores())</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(parallel)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(igraph)</span></code></pre></div>
<pre><code>## Warning: package &#39;igraph&#39; was built under R version 4.2.3</code></pre>
<pre><code>## 
## Attaching package: &#39;igraph&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     decompose, spectrum</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     union</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">library</span>(CITMIC)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="fu">detectCores</span>()</span></code></pre></div>
<pre><code>## [1] 12</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># Obtain the example data</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>GEP<span class="ot">&lt;-</span><span class="fu">GetData_CITMIC</span>(<span class="st">&quot;GEP&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co"># Calculate the Cell infiltration(lnScore)</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>lnScore<span class="ot">&lt;-</span><span class="fu">CITMIC</span>(GEP,<span class="at">weighted =</span> <span class="cn">TRUE</span>,<span class="at">base =</span> <span class="dv">10</span>,<span class="at">damping=</span><span class="fl">0.90</span>,<span class="at">cl.cores=</span><span class="dv">1</span>,<span class="at">cell.type=</span><span class="cn">NULL</span>)</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="fu">head</span>(lnScore)</span></code></pre></div>
<pre><code>##                        TCGA.EE.A2GJ.06A TCGA.EE.A2GI.06A TCGA.WE.A8ZM.06A
## Activated B cell              0.4155478        0.5319714        0.1549675
## Activated CD4+ T cells        0.6105840        0.6376140        0.3149371
## Activated CD8+ T cells        0.3987288        0.4678214        0.1503589
## aDCs                          0.9157299        0.9146414        0.8568350
## Adipocytes                    0.2190414        0.3170802        0.4874105
## Angiogenesis                  0.5762410        0.5273987        0.6974899
##                        TCGA.DA.A1IA.06A TCGA.D3.A51H.06A
## Activated B cell              0.1314579        0.7365269
## Activated CD4+ T cells        0.4463615        0.7212100
## Activated CD8+ T cells        0.2170825        0.5600082
## aDCs                          0.9070054        0.9060217
## Adipocytes                    0.3171054        0.2778043
## Angiogenesis                  0.5255911        0.5043280</code></pre>
<hr />
</div>
<div id="refenrence" class="section level2">
<h2>Refenrence</h2>
<p><a href="#https://pubmed.ncbi.nlm.nih.gov/24138885/">1. Bindea, G.,
et al., Spatiotemporal dynamics of intratumoral immune cells reveal the
immune landscape in human cancer. Immunity, 2013. 39(4): p. 782-95.</a>
<a href="#https://pubmed.ncbi.nlm.nih.gov/28052254/">2. Charoentong, P.,
et al., Pan-cancer Immunogenomic Analyses Reveal
Genotype-Immunophenotype Relationships and Predictors of Response to
Checkpoint Blockade. Cell Rep, 2017. 18(1): p. 248-262.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/28239471/">3. Danaher, P., et
al., Gene expression markers of Tumor Infiltrating Leukocytes. J
Immunother Cancer, 2017. 5: p. 18.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/28104840/">4. Davoli, T., et al.,
Tumor aneuploidy correlates with markers of immune evasion and with
reduced response to immunotherapy. Science, 2017. 355(6322).</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/30594216/">5. He, Y., et al.,
Classification of triple-negative breast cancers based on Immunogenomic
profiling. J Exp Clin Cancer Res, 2018. 37(1): p. 327.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/25594174/">6. Rooney, M.S., et
al., Molecular and genetic properties of tumors associated with local
immune cytolytic activity. Cell, 2015. 160(1-2): p. 48-61.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/27124452/">7. Tirosh, I., et al.,
Dissecting the multicellular ecosystem of metastatic melanoma by
single-cell RNA-seq. Science, 2016. 352(6282): p. 189-96.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/27765066/">8. Becht, E., et al.,
Estimating the population abundance of tissue-infiltrating immune and
stromal cell populations using gene expression. Genome Biol, 2016.
17(1): p. 218.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/29130882/">9. Racle, J., et al.,
Simultaneous enumeration of cancer and immune cell types from bulk tumor
gene expression data. Elife, 2017. 6.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/32274301/">10. Miao, Y.R., et
al., ImmuCellAI: A Unique Method for Comprehensive T-Cell Subsets
Abundance Prediction and its Application in Cancer Immunotherapy. Adv
Sci (Weinh), 2020. 7(7): p. 1902880.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/30127393/">11. Jiang, P., et al.,
Signatures of T cell dysfunction and exclusion predict cancer
immunotherapy response. Nat Med, 2018. 24(10): p. 1550-1558.</a> <a href="#https://pubmed.ncbi.nlm.nih.gov/29141660/">12. Aran, D., Z. Hu,
and A.J. Butte, xCell: digitally portraying the tissue cellular
heterogeneity landscape. Genome Biol, 2017. 18(1): p. 220.</a></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
